TestTools/StreamChatTestTools/Mocks/StreamChat/WebSocketClient/EventMiddlewares/NotificationExtensionLifecycle_Mock.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  NotificationExtensionLifecycle_Mock.swift
+//  StreamChatTests
+//
+//  Created by Pol Quintana on 7/2/23.
+//  Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+@testable import StreamChat
+
+final class NotificationExtensionLifecycle_Mock: NotificationExtensionLifecycle {
+
+    var mockIsAppReceivingWebSocketEvents: Bool?
+    var receivedIsReceivingEvents: Bool?
```

Comment:
Kinda disagree here. It is true that we should not be getting unexpected access. But I am a believer that a test suite should not crash under these circumstances.
Our current test suite tries to use ways to avoid crashes and instead report errors.